pipelines/deployer/deployer.yaml

```diff
@@ -1154,3 +1154,26 @@ jobs:
       action: destroy
     get_params:
       action: destroy
+  - task: final-cleanup
+    image: task-toolbox
+    config:
+      platform: linux
+      params:
+        ACCOUNT_ROLE_ARN: ((account-role-arn))
+        CLUSTER_NAME: ((cluster-name))
+      run:
+        path: /bin/bash
+        args:
+        - -eu
+        - -c
+        - |
+          echo "assuming aws deployer role..."
+          AWS_CREDS="$(aws-assume-role $ACCOUNT_ROLE_ARN)"
+          eval "${AWS_CREDS}"
+          echo "cleaning up orphaned log groups..."
+          aws logs delete-log-group --log-group-name /aws/containerinsights/${CLUSTER_NAME}/application
```

Comment:
when this script encounters a log groups that already does not exist, it will fail as the `delete-log-group` command will exit with an error causing the script to stop.

we should make the script capable of handling the case when the log groups don't exist.

You can achieve this by adding `|| echo 'already deleted'` to the end of each of the `aws` command lines to tell the script to just echo something when the command errors

Reply:
But that will swallow all the other errors that might come up in the aws command...

Reply:
@blairboy362 true of course ... but the alternative (iterating and filtering the log groups programatically) seemed like a lot more of an ask for what appears to be a bit of a hack around some other problem to do with log-groups not being managed via terraform correctly for on-demand clusters.

I'd be happy with either, just pointing out that it will currently break on the first log group that doesn't exist, and they usually _don't_ exist, so this will break most of the time on the first line.

Reply:
> log-groups not being managed via terraform correctly for on-demand clusters

I don't understand why on-demand clusters are different in this regard.

> they usually _don't_ exist, so this will break most of the time on the first line

If that's true then I agree. But make the `echo` something more meaningful like `echo 'aws exited with error. Assuming log group does not exist.'` Or put a `set +e` around the `aws` commands.

Reply:
They aren't "different" ... we just don't destroy any other clusters, so the issue is only present for on-demand cluster, which we destroy and then re-create again.... I _think_ the real issue is that some log groups are created by AWS as-required which conflicts with terraform managed groups ... so we should probably just removed the CreateLogGroup permission from everything and ensure we terraform them

Reply:
:+1: Indeed.

Reply:
```suggestion
          aws logs delete-log-group --log-group-name /aws/containerinsights/${CLUSTER_NAME}/application || echo 'aws exited with error. Assuming log group does not exist.'
```